### DISCUSSION TITLE
Экспорт плейлистов с id трека

### DISCUSSION THREAD

**Discussion by just-aa** [Chimildic/YaMuTools#62] [Идея]
Здравствуйте, уважаемые разработчики!
В свете исхода Sony Music из России, встал вопрос сохранения библиотеки ЯМузыки.

Добавьте, пожалуйста, в экспортёр опцию экспорта плейлиста с указанием id трека (realId)
Это поможет для восстановления мета-данных скаченных с ЯМузыки файлов: например, tag editor от Aimp умеет искать информацию для ID3v2 через интернет по названию трека.

Сопоставив скачанные файлы ЯМузыки с плейлистом с id-треков, можно, например, силами Advanced Renamer, быстро переименовать все треки :)

**Comment by Chimildic** [comment]
Здравствуйте, разработчик, к слову, один)

Если вы собираетесь скачивать треки с Яндекс.Музыки, то для чего все это нужно? Попробовал расширение _skyload_, скаченный файл уже имеет атрибуты альбома и исполнителя.

![image](https://user-images.githubusercontent.com/40138097/189176337-f415e138-939d-4016-86cf-b3f6fe103c5a.png)

**Reply by kisoid** [reply]
skyload не алё...

**Reply by Chimildic** [reply]
В каком смысле?

**Reply by just-aa** [reply]
Здравствуйте, @Chimildic !
Спасибо за быстрый ответ и ваше расширение :)

Для чего понадобился айди трека? Приложения ЯМузыка для Windows скачивает треки в формате .mp3, но вместо имени там realid (просто набор цифр). Имея на руках список треков с айди и именем файла, можно быстро восстановить все названия (сопоставить через, например, Advanced Renamer). Это удобнее, чем сторонние расширения, которые, чаще всего, позволяют качать треки по одному.

А так получается, что приложение ЯндексМузыка само скачало треки, которые надо только переименовать правильно.
<strike>Это даже не пиратство :)</strike>

**Reply by kisoid** [reply]
> В каком смысле?

https://addons.mozilla.org/ru/firefox/addon/skyload-media-downloader/?display=ru

> Если вы перешли по ссылке с другого сайта за расширением или темой, то этот продукт более недоступен. Это может быть из-за того, что:
> - Разработчик удалил его. Обычно разработчики делают это, если они больше не поддерживают расширение или тему, или заменили его.
> - Mozilla удалила его. Это могло произойти, если во время проверки расширения или темы были обнаружены проблемы, или если расширение или тема нарушает правила и условия addons.mozilla.org. Разработчик имеет возможность устранить проблемы и сделать дополнение доступным снова.

http://skyload.io/install/chrome

> В данный момент проходит модерация расширения Skyload в Chrome Web Store. Это может занять от пару часов до нескольких дней.

**Reply by Chimildic** [reply]
@kisoid доступно для оперы и яндекс браузера [отсюда](https://addons.opera.com/ru/extensions/details/skyload/?display=ru)  или ручной установки в хром через [руководство](https://vk.com/away.php?to=https%3A%2F%2Fwww.hashtap.com%2Fp%2FAPl6JR7ZrAlR&post=-59648048_2401&cc_key=)

@just-aa какой формат названия вам нужен?

**Reply by kisoid** [reply]
Для полного зоопарка только этих браузеров не хватает...
